### DISCUSSION TITLE
How do we want to handle page titles?

### DISCUSSION THREAD

**Discussion by stevepolitodesign** [thoughtbot/suspenders#1297] [General]
We currently rely on [title](https://github.com/calebhearth/title) to generate the `<title>`. However, with the introduction of https://github.com/rails/rails/pull/49702, do we still feel we need this dependency?

If we want to continue using [title](https://github.com/calebhearth/title), I think we'll need to ensure the existing [regex](https://github.com/thoughtbot/suspenders/blob/7c74d88052d8e9b0de69dd178a3425ed359ae850/lib/generators/suspenders/views_generator.rb#L33) still works.

**Comment by louis-antonopoulos** [comment]
If Suspenders continues using the Title gem, it would be helpful for the app to add the `titles/application` keys to `en.yml` so that the user knows where that value is coming from, as demonstrated in the Title [README](https://github.com/calebhearth/title?tab=readme-ov-file#usage).

Without the keys in `en.yml`, Title defaults to something (possibly the project name), and the string value is not searchable in the project to know where to change it. You have to look at the application layout, see the `title` reference, then notice that  `title` is in the `Gemfile`, then read their `README` to figure out how to set it.

**Suggested addition:**

```
en:
  hello: 'Hello world'
  titles:
    application: Hello World
```

@stevepolitodesign Do you think a separate issue about this would be helpful, or is it better to keep it contained here for the moment until we decide whether or not to use Title?

**Comment by stevepolitodesign** [comment]
I personally think that in the spirit of keeping Suspenders as close to `rails new` as possible, we should remove our dependency on [title](https://github.com/calebhearth/title).

Let's see what others have to say, and if we find that we want to keep our dependency with [title](https://github.com/calebhearth/title), then we can close this issue and open a new one with your suggested improvements.

**Comment by stevepolitodesign** [comment]
Noting that in the latest release, we removed the title gem.
